store/sqlstore/emoji_store.go

```diff
@@ -80,47 +105,40 @@ func (es SqlEmojiStore) Get(id string, allowFromCache bool) (*model.Emoji, *mode
 		}
 	}
 
-	var emoji *model.Emoji
-
-	if err := es.GetReplica().SelectOne(&emoji,
-		`SELECT
-			*
-		FROM
-			Emoji
-		WHERE
-			Id = :Id
-			AND DeleteAt = 0`, map[string]interface{}{"Id": id}); err != nil {
-		return nil, model.NewAppError("SqlEmojiStore.Get", "store.sql_emoji.get.app_error", nil, "id="+id+", "+err.Error(), http.StatusNotFound)
-	}
-
-	if allowFromCache {
-		emojiCache.AddWithExpiresInSecs(id, emoji, EMOJI_CACHE_SEC)
+	if emoji, err := es.GetBy("Id", id); err == nil {
+		if allowFromCache {
+			emojiCache.AddWithExpiresInSecs(id, emoji, EMOJI_CACHE_SEC)
+		}
+		return emoji, nil
+	} else {
+		return nil, err
 	}
-
-	return emoji, nil
 }
 
-func (es SqlEmojiStore) GetByName(name string) (*model.Emoji, *model.AppError) {
-
-	var emoji *model.Emoji
-
-	if err := es.GetReplica().SelectOne(&emoji,
-		`SELECT
-			*
-		FROM
-			Emoji
-		WHERE
-			Name = :Name
-			AND DeleteAt = 0`, map[string]interface{}{"Name": name}); err != nil {
+func (es SqlEmojiStore) GetByName(name string, allowFromCache bool) (*model.Emoji, *model.AppError) {
+	if id, ok := model.GetSystemEmojiId(name); ok {
+		return es.Get(id, allowFromCache)
+	}
 
-		if err == sql.ErrNoRows {
-			return nil, model.NewAppError("SqlEmojiStore.GetByName", "store.sql_emoji.get_by_name.app_error", nil, "name="+name+", "+err.Error(), http.StatusNotFound)
+	if allowFromCache {
+		for key := range emojiCache.Keys() {
+			if item, ok := emojiCache.Get(key); ok {
+				emoji := item.(*model.Emoji)
+				if emoji.Name == name {
+					if es.metrics != nil {
+						es.metrics.IncrementMemCacheHitCounter("Emoji")
+					}
+					return emoji, nil
+				}
+			}
 		}
 
-		return nil, model.NewAppError("SqlEmojiStore.GetByName", "store.sql_emoji.get_by_name.app_error", nil, "name="+name+", "+err.Error(), http.StatusInternalServerError)
+		if es.metrics != nil {
+			es.metrics.IncrementMemCacheMissCounter("Emoji")
```

Comment:
@jespino, I know you had some thoughts on when to increment the miss counter. Should we miss even if `!allowFromCache`?